### PR TITLE
Run mission status request in separate thread

### PR DESCRIPTION
**Pull request by vetlek**

### CODE REVIEW

File: src/isar/state_machine/states/monitor.py [hunk 1]
```diff
@@ -12,6 +13,11 @@
     from isar.state_machine.state_machine import StateMachine
 
 
+@dataclass
+class MissionStatusThreadOutput:
```
Comment (oysand) [inline src/isar/state_machine/states/monitor.py:17]: Should this dataclass be placed in the models folder and imported from there?

File: src/isar/state_machine/states/monitor.py [hunk 2]
```diff
@@ -30,37 +39,54 @@ def start(self):
...
-                break
-
             if self.state_machine.should_send_status():
                 self.state_machine.send_status()
-            time.sleep(self.state_machine.sleep_time)
+
+            if not self.mission_status_thread:
+                self.mission_status_thread = threading.Thread(
+                    target=mission_status_thread,
+                    args=(
```
Comment (oysand) [inline src/isar/state_machine/states/monitor.py:60]: Could we list which input each of these would be, for instance: mission_status_func=self.state_machine.robot.mission_status?

File: src/isar/state_machine/states/monitor.py [hunk 3]
```diff
@@ -86,3 +112,12 @@ def _log_status(self, mission_status: MissionStatus):
                 current_step=self.state_machine.status.current_mission_step,
             )
         self.iteration_counter += 1
+
+
+def mission_status_thread(
```
Comment (oysand) [inline src/isar/state_machine/states/monitor.py:117]: Would it be better to have this function as part of the class?

File: src/isar/state_machine/states/monitor.py [hunk 4]
```diff
@@ -30,37 +39,54 @@ def start(self):
 
     def stop(self):
         self.iteration_counter = 0
+        if self.mission_status_thread:
+            self.mission_status_thread.join()
```
Comment (oysand) [inline src/isar/state_machine/states/monitor.py:43]: Could we risk the code hanging on this step if something is holding up the thread?

**Review by oysand — DISMISSED** [review]
See comments/questions, large overlap with PR #70

File: src/isar/state_machine/states/monitor.py (hunk 4 above)
Comment (vetlek) [reply src/isar/state_machine/states/monitor.py:43]: The code will hang if something is holding up the thread. It is here to make sure we don't have any threads going when we leave the state and make sure everything is closed properly. Could possibly kill the thread in some way, but I think it is bad practice. There should be some sort of timeout in the request module to limit the time we hang here, so should be a limited time in the worst case

File: src/isar/state_machine/states/monitor.py (hunk 3 above)
Comment (sondreo) [reply src/isar/state_machine/states/monitor.py]: Implemented.

**Review by oysand — APPROVED** [review]
LGTM

**Pull request merged**